src/System.Reflection.Metadata/src/System/Reflection/PortableExecutable/PEHeaderBuilder.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Reflection.PortableExecutable
+{
+    public sealed class PEHeaderBuilder
```

Comment:
We talked about this one offline being immutable but named -Builder. I think PEHeaderData was the name I liked best.